jme3-core/src/main/java/com/jme3/anim/MorphTrack.java

```diff
@@ -101,10 +101,13 @@ public void setKeyframesWeight(float[] weights){
         if (weights.length == 0) {
             throw new RuntimeException("MorphTrack with no weight keyframes!");
         }
+        if(times.length * nbMorphTargets != weights.length){
+            throw new RuntimeException("weights.length must equal nbMorphTargets * times.length");
```

Comment:
The exception type can be more specific. I advise using `IllegalArgumentException`. I encourage you to also improve line 102 along those lines.

Reply:
Agreed - I wrote this out with `IllegalArgumentException` at first but changed to the current `RuntimeException` since that's what's currently used. `TransformTrack` also similarly misuses `RuntimeException`, but since the change will have little to no effect on existing exception handlers I think we should change all similar uses to `IllegalArgumentException`.